src/main/java/com/iexec/sms/tee/session/scone/cas/SconeSession.java

```diff
@@ -85,8 +85,15 @@ public static class Security {
         @JsonProperty("attestation")
         private Attestation attestation;
 
-        public Security(List<String> tolerate, List<String> ignoreAdvisories) {
-            this.attestation = new Attestation(tolerate, ignoreAdvisories);
+        public Security(List<String> tolerate,
+                        List<String> ignoreAdvisories,
+                        String mode,
+                        String url)
+                throws IllegalArgumentException {
+            this.attestation = new Attestation(tolerate, ignoreAdvisories, mode, url);
+            if ("maa".equals(mode) && url == null) {
+                throw new IllegalArgumentException("Attestation URL can not be null when scone session mode is 'maa'");
+            }
```

Comment:
I very much like the idea but not the place it is located.
What will happen when this exception is triggered ?
Would there be a better place to do this check ?

Reply:
Is it best to use it at the base of the potential problem (in the `Attestation` constructor) or at the end (in the `SconeSessionMakerService` class, inside the method that use the `Security` constructor)? 

I'd say at the end, maybe?

Reply:
Check moved to `SconeSessionSecurityConfig` :+1: